src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,49 +1,79 @@
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+import { Todo } from '../../types/Todo';
 
-export const TodoModal: React.FC = () => {
-  return (
-    <div className="modal is-active" data-cy="modal">
-      <div className="modal-background" />
-
-      {true ? (
-        <Loader />
-      ) : (
-        <div className="modal-card">
-          <header className="modal-card-head">
-            <div
-              className="modal-card-title has-text-weight-medium"
-              data-cy="modal-header"
-            >
-              Todo #2
-            </div>
+type Props = {
+  todo: Todo | undefined,
+  onClose: () => void,
+};
+
+export const TodoModal: React.FC<Props> = React.memo(
+  ({
+    todo,
+    onClose,
+  }) => {
+    const [users, setUsers] = useState<User>();
```

Comment:
Rename 'users' state to 'user' and its corresponding setState function to 'setUser' since it's holding a single user object, not an array of users.